rs/bitcoin/validation/src/header.rs

```diff
@@ -90,82 +71,14 @@ pub fn validate_header(
     Ok(())
 }
 
-/// Checks if block height is higher than the last checkpoint height.
-/// By beeing beyond the last checkpoint we are sure that we store the correct chain up to the height
-/// of the last checkpoint.  
-pub fn is_beyond_last_checkpoint(network: &Network, height: BlockHeight) -> bool {
-    match last_checkpoint(network) {
-        Some(last) => last <= height,
-        None => true,
-    }
-}
-
-/// This validates the header against the network's checkpoints.
-/// 1. If the next header is at a checkpoint height, the checkpoint is compared to the next header's block hash.
-/// 2. If the header is not the same height, the function then compares the height to the latest checkpoint.
-///    If the next header's height is less than the last checkpoint's height, the header is invalid.
-fn is_checkpoint_valid(
-    network: &Network,
-    prev_height: BlockHeight,
-    header: &BlockHeader,
-    chain_height: BlockHeight,
-) -> bool {
-    let checkpoints = checkpoints(network);
-    let next_height = prev_height.saturating_add(1);
-    if let Some(next_hash) = checkpoints.get(&next_height) {
-        return *next_hash == header.block_hash();
-    }
-
-    let checkpoint_height = latest_checkpoint_height(network, chain_height);
-    next_height > checkpoint_height
-}
-
-/// This validates that the header has a height that is within 1 year of the tip height.
-fn is_header_within_one_year_of_tip(prev_height: BlockHeight, chain_height: BlockHeight) -> bool {
-    // perhaps checked_add would be preferable here, if the next height would cause an overflow,
-    // we should know about it instead of being swallowed.
-    let header_height = prev_height
-        .checked_add(1)
-        .expect("next height causes an overflow");
-
-    let height_one_year_ago = chain_height.saturating_sub(BLOCKS_IN_ONE_YEAR);
-    header_height >= height_one_year_ago
-}
-
-/// Validates if a header's timestamp is valid.
-/// Bitcoin Protocol Rules wiki https://en.bitcoin.it/wiki/Protocol_rules says,
-/// "Reject if timestamp is the median time of the last 11 blocks or before"
-fn is_timestamp_valid(store: &impl HeaderStore, header: &BlockHeader) -> bool {
```

Comment:
This is the only check I am a bit hesitant to keep since it is documented as part of the BTC protocol rules

https://en.bitcoin.it/wiki/Protocol_rules

Do you know if testnet and testnet4 follow this particular check ?